server/datastore/mysql/software_installers.go

```diff
@@ -73,30 +73,35 @@ func (ds *Datastore) GetSoftwareInstallDetails(ctx context.Context, executionId
 func (ds *Datastore) MatchOrCreateSoftwareInstaller(ctx context.Context, payload *fleet.UploadSoftwareInstallerPayload) (uint, error) {
 	titleID, err := ds.getOrGenerateSoftwareInstallerTitleID(ctx, payload)
 	if err != nil {
-		return 0, err
+		return 0, ctxerr.Wrap(ctx, err, "get or generate software installer title ID")
```

Comment:
Why do these need to be wrapped? Doesn't the above method generate a proper error?

Reply:
In this case I needed to trace which func in the method was returning errors